tests/unit/eblob_wrapper.h

```diff
@@ -0,0 +1,120 @@
+#pragma once
+
+#include <boost/filesystem.hpp>
+
+#include <library/blob.h>
+
+#include "library/crypto/sha512.h"
+
+#include <eblob/eblob.hpp>
+
+#include <vector>
+
+
+class Item {
+public:
+	Item(int key_, const eblob_key &hashed_key_, const std::vector<char> value_)
+	: key(key_)
+	, value(std::move(value_))
+	{
+		memcpy(hashed_key.id, hashed_key_.id, EBLOB_ID_SIZE);
+	}
+
+	bool checked = false;
+	bool removed = false;
+	int key;
+	eblob_key hashed_key;
+	std::vector<char> value;
+
+	bool operator< (const Item &rhs) const {
+		return eblob_id_cmp(hashed_key.id, rhs.hashed_key.id) < 0;
+	}
+};
+
+
+class eblob_wrapper {
+public:
+	eblob_wrapper()
+	: data_dir_template_("/tmp/eblob-test-XXXXXX")
+	, data_dir_{mkdtemp(&data_dir_template_.front())}
+	, data_path_{data_dir_ + "/data"}
+	, log_path_{data_dir_ + "/log.log"}
+	, logger_{log_path_.c_str(), EBLOB_LOG_DEBUG} {
+		default_config.blob_flags = EBLOB_L2HASH | EBLOB_DISABLE_THREADS | EBLOB_AUTO_INDEXSORT;
+		default_config.sync = -2;
+		default_config.log = logger_.log();
+		default_config.file = (char *)data_path_.c_str();
+		default_config.blob_size = EBLOB_BLOB_DEFAULT_BLOB_SIZE;
+		default_config.records_in_blob = EBLOB_BLOB_DEFAULT_RECORDS_IN_BLOB;
+		default_config.defrag_percentage = EBLOB_DEFAULT_DEFRAG_PERCENTAGE;
+		default_config.defrag_timeout = EBLOB_DEFAULT_DEFRAG_TIMEOUT;
+		default_config.index_block_size = EBLOB_INDEX_DEFAULT_BLOCK_SIZE;
+		default_config.index_block_bloom_length = EBLOB_INDEX_DEFAULT_BLOCK_BLOOM_LENGTH;
+		default_config.blob_size_limit = UINT64_MAX;
+		default_config.defrag_time = EBLOB_DEFAULT_DEFRAG_TIME;
+		default_config.defrag_splay = EBLOB_DEFAULT_DEFRAG_SPLAY;
+		default_config.periodic_timeout = EBLOB_DEFAULT_PERIODIC_THREAD_TIMEOUT;
+		default_config.stat_id = 12345;
+		default_config.chunks_dir = (char*)data_dir_.c_str();
+	}
+
+	void start(eblob_config *config = nullptr) {
+		if (config != nullptr)
+			backend_ = eblob_init(config);
+		else
+			backend_ = eblob_init(&default_config);
+	}
+
+	void restart(eblob_config *config = nullptr) {
+		stop();
+		start(config);
+	}
+
+	void stop() {
+		if (backend_) {
+			eblob_cleanup(backend_);
+			backend_ = nullptr;
+		}
+	}
+
+	~eblob_wrapper() {
+		stop();
+		boost::filesystem::remove_all(data_dir_);
+	}
+
+	eblob_backend *get() { return backend_; }
+	const eblob_backend *get() const { return backend_; }
+
+	Item generate_item(int key, const std::vector<char> &data) {
```

Comment:
These `*_item` methods have been appended to common class `eblob_wrapper`, but the class is used also in corruption_test, where `*_item` methods aren't used. Better to transform `*_item` methods to free functions, and then better move them to defrag_test since they used only there.

Reply:
Later someone may want to use this methods. They are general slightly.